src/Devtools.hs

```diff
@@ -1,31 +1,59 @@
-module Devtools (Config(..), defaultConfig, defaultMain, main, testTree) where
+module Devtools
+  ( Config(..)
+  , FormatterOptions(..)
+  , defaultConfig
+  , defaultFomatterOptions
+  , defaultMain
+  , main
+  , testTree
+  )
+where
 
-import Devtools.Prelude
-import System.IO (putStrLn)
+import           Devtools.Files
+import           Devtools.Prelude
 
-import qualified Devtools.Dependencies as Dependencies
-import qualified Devtools.HLint        as HLint
-import qualified Test.Tasty            as Tasty
+import qualified Data.List.NonEmpty            as NE
+import qualified Devtools.Brittany             as Brittany
+import qualified Devtools.Dependencies         as Dependencies
+import qualified Devtools.HLint                as HLint
+import qualified System.Path                   as Path
+import qualified Test.Tasty                    as Tasty
 
-newtype Config = Config
-  { hlintArguments :: [String]
+data Config = Config
+  { hlintArguments   :: [String]
+  , formatterOptions :: FormatterOptions
   }
 
-defaultConfig :: Config
-defaultConfig = Config
-  { hlintArguments = []
+data FormatterOptions = FormatterOptions
+  { configFile    :: Maybe Path.RelFile
+  , directories   :: NE.NonEmpty Path.RelDir
+  , filesToIgnore :: [Path.RelFile]
```

Comment:
There are cases where even formatting inbuilt inline ignore annotations/comments don't help in ensuring proper formatting that doesn't lead to module breakage. In such cases we bypassing formatting such files entirely

E.g GADTs formatting issue in Brittany https://github.com/lspitzner/brittany/issues/242